### ISSUE TITLE
can we support IsolatedModules by default?

### ISSUE THREAD

**Issue by dmrusyniak** [TrueFiEng/useDApp#736]
can we support IsolatedModules by default?

since upgrading to 1.0.1 I'm receiving build errors in my Typescript Next.js project, where I need to have the isolatedModules flag provided in our tsconfig.json

they look like: 

`Type error: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.`

we did some digging and discovered that simply updating the exports in packages/core/src/index.ts to explicitly export the types as types, will resolve the Type error and allow the module to build again. 

for example, in the exports from './constants', we can do this: 

```
export type {
  Chain,
  Config,
  FullConfig,
  MulticallAddresses,
  NodeUrls,
} from './constants'

export {
  ChainId,
  BNB,
  BUSD,
  DEFAULT_SUPPORTED_CHAINS,
  Dai,
  ERC20,
  ERC20Interface,
  ERC20Mock,
  ERC20MockInterface,
  Ether,
  KovanDai,
  KovanEther,
  MultiCall,
  MultiCall2,
  MultiCall2ABI,
  MultiCallABI,
  NATIVE_CURRENCY,
  TestBNB,
  TestBUSD,
} from './constants'
```

we'll need to apply this export change to all type exports, and I think this will resolve our issue.

**Comment by dmrusyniak** [comment]
I'm not sure I know how you would do that or perhaps I don't understand the question?

I'm happy to make the PR and have it merged, as it would really unblock my team-- but when I tried making the PR I got a permissions issue? 

Any idea how we can move this forward?

Your help is greatly appreciated!

**Comment by rzadp** [comment]
> @rzadp is this file auto-generated? What do you mean by automatically?

By automation I meant that it is enforced that we do `export type` for all types - but it's as simple as enabling the `IsolatedModules` in our config. 

https://github.com/TrueFiEng/useDApp/pull/762 is on the way.

**Comment by rzadp** [comment]
@dmrusyniak @jemilezzet The dev version of usedapp containing this change is released on NPM under `@usedapp/core@1.0.3-dev.1536229`

Let me know if that resolves your problems.
